### PR TITLE
add authorize admin user policy and test them

### PR DESCRIPTION
**Pull request by VickTor61**

### CODE REVIEW

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
+module Admin
+  class UsersController < ApplicationController
+
+    def index
+      @users = User.all
+      authorize @users
```
Comment (EnomaDebby) [inline app/controllers/admin/users_controller.rb:6]: This is not the right way to do this. Did you check the bundit documentation?

File: app/controllers/admin/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    end
+
+    def show
+      @user = authorize User.find(params[:id])
```
Comment (EnomaDebby) [inline app/controllers/admin/users_controller.rb]: Get the user first and call authorize on the user

File: app/controllers/admin/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    end
+
+    private
+
+    def user_params
+      params.require(:user).permit(:email, :password, :password_confirmation)
+    end
+
+  end
+end
```
Comment (EnomaDebby) [inline app/controllers/admin/users_controller.rb]: no new line

File: app/models/user.rb [hunk 4]
```diff
@@ -7,4 +7,8 @@ class User < ApplicationRecord
   has_many :campaigns
   has_many :user_roles
   has_many :roles, through: :user_roles
+
+  def admin?
+    roles.where(slug: 'admin').any?
+  end
 end
```
Comment (EnomaDebby) [inline app/models/user.rb]: new line

File: app/policies/user_policy.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  #   # NOTE: Be explicit about which records you allow access to!
+  #   # def resolve
+  #   #   scope.all
+  #   # end
+  # end
+
+  def index?
+    user.admin?
+  end
+
```
Comment (EnomaDebby) [inline app/policies/user_policy.rb:13]: extra spaces

File: app/policies/user_policy.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+  def show?
+    user.admin?
+  end
+
+  def new?
+    user.admin?
+  end
```
Comment (EnomaDebby) [inline app/policies/user_policy.rb:20]: no space between the next action. Please indent properly

File: app/views/admin/users/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,39 @@
+<%= form_with model: user, local: true do |form| %>
+
+  <% if @user.errors.any? %>
+    <div id="error_explanation">
+      <!--      <h2><%#= pluralize(@user.errors.count, "error") %> prohibited this comment from being saved:</h2>-->
```
Comment (EnomaDebby) [inline app/views/admin/users/_form.html.erb]: remove if not needed

File: app/views/admin/users/_form.html.erb [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  <% end %>
+
+
+
+  <div class="form-group-sm p-2">
+    <%= form.label :email %><br>
+    <%= form.text_field :email, class: 'form-control'%>
+  </div>
+
+
```
Comment (EnomaDebby) [inline app/views/admin/users/_form.html.erb:22]: indent this file properly. No extra spaces

File: app/views/admin/users/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+      <% if policy(user).show? %>
+        <%= link_to user.email, admin_user_path(user.id), class: "link-info text-uppercase text-decoration-none" %>
+      <% end %>
+
+    <div class="d-flex align-items-center ">
+      <% if policy(user).destroy? %>
+        <p style="margin: 0"><%= link_to "Destroy", admin_user_path(user), method: :delete, class: "btn btn-danger align-self-center" %></p>
+      <% end %>
+
+      <% if policy(user).update? %>
```
Comment (EnomaDebby) [inline app/views/admin/users/index.html.erb:21]: Is there no edit policy?

File: app/views/admin/users/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,7 @@
+<div class="card" style="width: 30rem; margin: 3rem auto; ">
+  <h1 class="card-header card-title"> New user </h1>
+
+
```
Comment (EnomaDebby) [inline app/views/admin/users/new.html.erb]: format correctly

File: spec/policies/user_policy_spec.rb [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+    end
+    it 'allows access to admin users' do
+      expect(subject).to permit(admin.user)
+    end
+  end
+
+  permissions :new? do
+    it 'denies access to non admin users' do
+      expect(subject).not_to permit(user)
+    end
```
Comment (EnomaDebby) [inline spec/policies/user_policy_spec.rb:30]: indent properly

**Review by EnomaDebby — CHANGES_REQUESTED** [review]

File: app/controllers/admin/users_controller.rb (hunk 1 above)
Comment (VickTor61) [reply app/controllers/admin/users_controller.rb:6]: Yes ma that's what the documentation says

File: app/views/admin/users/index.html.erb (hunk 9 above)
Comment (VickTor61) [reply app/views/admin/users/index.html.erb:21]: The update policy covers for the edit.

**Review by EnomaDebby — APPROVED** [review]

**Pull request merged**